src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -125,9 +125,11 @@ public void OnStartActivity(Activity activity, object payload)
         if (textMapPropagator is not TraceContextPropagator)
         {
             var ctx = textMapPropagator.Extract(default, request, HttpRequestHeaderValuesGetter);
-
+            var activityContext = new ActivityContext(activity.TraceId, activity.ParentSpanId, activity.ActivityTraceFlags, activity.TraceStateString, true);
             if (ctx.ActivityContext.IsValid()
-                && ctx.ActivityContext != new ActivityContext(activity.TraceId, activity.ParentSpanId, activity.ActivityTraceFlags, activity.TraceStateString, true))
+                && ((ctx.ActivityContext.TraceId != activityContext.TraceId)
```

Comment:
could we not compare like below?

```
          if (ctx.ActivityContext.IsValid()
              && (!ctx.ActivityContext.TraceId.Equals(activity.TraceId) ||
              !ctx.ActivityContext.SpanId.Equals(activity.SpanId) ||
              !ctx.ActivityContext.TraceState.Equals(activity.TraceStateString)))
```

Reply:
The double negations confuse me tbh.
Wouldn't you rather do something like

```csharp
if (ctx.ActivityContext.IsValid()
  && !((ctx.ActivityContext.TraceId == activity.TraceId) &&
  (ctx.ActivityContext.SpanId == activity.ParentSpanId) &&
  (ctx.ActivityContext.TraceState == activity.TraceStateString)))
```

Reply:
My snippet had a typo, Fixed it. Your suggestion also looks good to me.